Autopilot/PathManager/Inc/cruisingMode/cruisingModeStages.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef CRUISING_MODE_STAGES_HPP
+#define CRUISING_MODE_STAGES_HPP
+
+#include "cruisingModeStageManager.hpp"
+#include "cruisingMode.hpp"
+
+#include "waypointManager.hpp"
+#include "TelemPathInterface.hpp"
+#include "AutoSteer.hpp"
+#include "cruisingFlight.hpp"
+
+class CruisingFlight : public CruisingModeStageManager {
+    public:
+        void enter(CruisingMode* cruise_mode) { (void) cruise_mode; }
+        void execute(CruisingMode* cruise_mode);
+        void exit(CruisingMode* cruise_mode) { (void) cruise_mode; }
+        static CruisingModeStageManager& getInstance();
+
+    private: 
+        CruisingFlight() : in_hold {false}, going_home {false} {}
+        CruisingFlight(const CruisingFlight& other);
+        CruisingFlight& operator=(const CruisingFlight& other);
+
+        WaypointManager cruising_state_manager;
+        int waypoint_id_array[PATH_BUFFER_SIZE]; // Stores ids of the waypoints in the flight path in the order that they are executed
```

Comment:
stdint please

Reply:
I guess I can make them 16 bit. Don't want to risk the case where we overflow an 8 bit integer

Reply:
addressed